src/core/service.c

```diff
@@ -2644,6 +2644,10 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
         else
                 assert_not_reached("Unknown code");
 
+        if (f != SERVICE_SUCCESS && status == SIGTERM &&
```

Comment:
Please add the comment indicating this is a hack and also provide bugzilla link.

Reply:
> Please add the comment indicating this is a hack and also provide bugzilla link.

OK, thanks.